bittrade_binance_websocket/rest/http_factory_decorator.py

```diff
@@ -9,14 +9,14 @@
 
 
 # TODO this typing does not work, it does not allow us to define the sub type of the response's result
-R = TypeVar("R", bound=models.HttpResponse)
+R = TypeVar("R")
 
-def http_factory(fn: Callable[P, models.RequestMessage]):
+def http_factory(fn: Callable[P, models.RequestMessage], return_type: Type[R]):
```

Comment:
Oh cool, I've been trying to figure out how to type; does this work?

Reply:
Yes it works, i believe i have an example